CrossGambling/Locale/Locale.lua

```diff
@@ -0,0 +1,63 @@
+        CrossGambling.L[".:Welcome to CrossGambling:. User's Roll - (%s) - Type 1 to Join (-1 to withdraw)"] = ".:Welcome to CrossGambling:. User's Roll - (%s) - Type 1 to Join (-1 to withdraw)"
+        CrossGambling.L["Last Call to join!"] = "Last Call to join!";
+        CrossGambling.L["%s still needs to roll."] = "%s still needs to roll.";
+        CrossGambling.L["Roll mow!"] = "Roll mow!";
+        CrossGambling.L["Not enough Players!"] = "Not enough Players!";
+        CrossGambling.L["High end tiebreaker! Roll 1- %s now!"] = "High end tiebreaker! Roll 1- %s now!";
+        CrossGambling.L["Low end tiebreaker! Roll 1- %s now!"] = "Low end tiebreaker! Roll 1- %s now!";
+        CrossGambling.L["%d.  %s %s %d total"] = "%d.  %s won %d total";
+        CrossGambling.L["%d.  %s %s %d total"] = "%d.  %s lost %d total";
+        CrossGambling.L["The house has taken %s total."] = "The house has taken %s total.";
+    elseif locale == "deDE" then
+        CrossGambling.L = CrossGambling.L or {}
+        CrossGambling.L["RollSplitStringIndexesPlayer"] = 1;
+        CrossGambling.L["RollSplitStringIndexesJunk"] = 2;
+        CrossGambling.L["RollSplitStringIndexesRoll"] = 4;
+        CrossGambling.L["RollSplitStringIndexesRange"] = 5;
+        CrossGambling.L["RollSplitJunkString"] = "würfelt.";
+        CrossGambling.L["%s owes %s %s gold!"] = "%s schuldet %s %s gold!";
+        CrossGambling.L["%s owes %s %s gold and %s gold the guild bank!"] = "%s schuldet %s %s gold und %s Gold der Gildenbank!";
+        CrossGambling.L["%s can now set the next gambling amount by saying !amount x"] = "%s kann nun den Einsatz der nächsten Spielrunde festlegen. - !amount x";
+        CrossGambling.L["It was a tie! No payouts on this roll!"] = "Unentschieden! Keine Auszahlungen auf diese Runde!";
+        CrossGambling.L[".:Welcome to CrossGambling:. User's Roll - (%s) - Type 1 to Join (-1 to withdraw)"] = ".:Willkommen bei CrossGambling:. Nutzer würfeln - (%s) - Tippe 1 zum Teilnehmen (-1 um nicht mehr Teilzunehmen.)"
+        CrossGambling.L["Last Call to join!"] = "Letzte Chance, um teilzunehmen!";
+        CrossGambling.L["%s still needs to roll."] = "%s muss noch würfeln.";
+        CrossGambling.L["Roll mow!"] = "Jetzt würfeln!";
+        CrossGambling.L["Not enough Players!"] = "Nicht genug Teilnehmer!";
+        CrossGambling.L["High end tiebreaker! Roll 1- %s now!"] = "High end tiebreaker! Roll 1- %s now!";
+        CrossGambling.L["Low end tiebreaker! Roll 1- %s now!"] = "Low end tiebreaker! Roll 1- %s now!";
+        CrossGambling.L["%d.  %s %s %d total"] = "%d.  %s hat insgesamt %d Gold gewonnen.";
+        CrossGambling.L["%d.  %s %s %d total"] = "%d.  %s hat insgesamt %d Gold verloren.";
+        CrossGambling.L["The house has taken %s total."] = "Das Haus hat insgesamt %s Gold bekommen.";
+    else
+        error("Unknown locale: "..tostring(locale))
```

Comment:
Well, for the period, we could simply make some regex no?  `.*\.?$` Matches both with and without a period at the end of the line.  What line does the rolling match happen?

Reply:
> Well, for the period, we could simply make some regex no? `.*\.?$` Matches both with and without a period at the end of the line. What line does the rolling match happen?

Yes maybe we must filter the RollSplitStringIndexesRange thats there ist the Result only (x-yyy) without the dot.

CrossGambling.lua - 668
In the function CrossGambling_ParseRoll(temp2)
```
local temp1 = strlower(temp2);
    local rollSplitTbl = {strsplit(" ", temp1)};
    local player = rollSplitTbl[L["RollSplitStringIndexesPlayer"]];
    local junk = rollSplitTbl[L["RollSplitStringIndexesJunk"]];
    local roll = rollSplitTbl[L["RollSplitStringIndexesRoll"]];
    local range = rollSplitTbl[L["RollSplitStringIndexesRange"]];
```
and in the Locale.lua
For enUS and enGB
```
        CrossGambling.L["RollSplitStringIndexesPlayer"] = 1;
        CrossGambling.L["RollSplitStringIndexesJunk"] = 2;
        CrossGambling.L["RollSplitStringIndexesRoll"] = 3;
        CrossGambling.L["RollSplitStringIndexesRange"] = 4;
```
For German
```
        CrossGambling.L["RollSplitStringIndexesPlayer"] = 1;
        CrossGambling.L["RollSplitStringIndexesJunk"] = 2;
        CrossGambling.L["RollSplitStringIndexesRoll"] = 4;
        CrossGambling.L["RollSplitStringIndexesRange"] = 5;
```

English: playername rolls 48 (1-100)
German: playername dices. Result: 48 (1-100)

I'll see if I find the Roll messages from all languages.

> The main thing I'm worried about is if an 'unsupported' language uses this, then the addon will stop working entirely.

I do not understand your concern, currently it is exactly like that. The current version on Curse / Twitch only works in English client language.

I think better if unsupported cleints just do not work. Instead of making it appear that it works. But you're right, it should actually be supported for all clients / languages.

But I think that you have to expand step by step. Or language by language :)